Implement new front end design

TammyTee opened the pull request:
## Summary

This resolves #1 🎉 

**Notable changes (some outside of FED)**

* Ran Headwind on the project to organize the tailwind classes
* Added a `skill_level` column to modules so that they can be organized on the listing page by beginner, intermediate, and advanced
* Changed the module listing page to list all module **NOT** in a users track when they're logged in
* Add filter to Modules in Nova to filter by skill level
* Add new layout that has no nav to use for auth views
* Organize Vue component imports for those that are used globally 


### Question(s)
* Is `learn.blade.php` still in use?

mattstauffer commented:
> Is learn.blade.php still in use?

Hm I'd lean toward no? But let's see lol

damiani commented on resources/views/partials/resource-on-module-page-with-border.blade.php:
```diff
@@ -0,0 +1,16 @@
+                :initial-is-completed="{{ $completedResources->contains($resource->id) ? 'true' : 'false' }}"
+                type="{{ $resource->getMorphClass() }}"
+                id="{{ $resource->id }}"
+                ></completed-checkbox>
```
```suggestion
            />
```

damiani commented on resources/views/partials/resource-on-module-page.blade.php:
```diff
@@ -1,11 +1,14 @@
+            :initial-is-completed="{{ $completedResources->contains($resource->id) ? 'true' : 'false' }}"
+            type="{{ $resource->getMorphClass() }}"
+            id="{{ $resource->id }}"
+            ></completed-checkbox>
```
```suggestion
        />
```

damiani commented on resources/views/vendor/nova/partials/footer.blade.php:
```diff
@@ -1,5 +1,5 @@
-<p class="mt-8 text-center text-xs text-80">
-    <a href="https://nova.laravel.com" class="text-primary dim no-underline">Laravel Nova</a>
+<p class="mt-8 text-xs text-center text-80">
+    <a href="https://nova.laravel.com" class="no-underline text-primary dim">Laravel Nova</a>
```
We don't need to make changes to Nova's vendor files

damiani commented on resources/views/vendor/nova/partials/user.blade.php:
```diff
@@ -14,7 +14,7 @@ class="rounded-full w-8 h-8 mr-3"
     <ul class="list-reset">
         <li>
-            <a href="{{ route('nova.logout') }}" class="block no-underline text-90 hover:bg-30 p-3">
+            <a href="{{ route('nova.logout') }}" class="block p-3 no-underline text-90 hover:bg-30">
```
We don't need to make changes to Nova's vendor files

TammyTee commented on app/Module.php:
```diff
@@ -13,6 +13,16 @@ class Module extends Model implements Completable
+        self::INTERMEDIATE_SKILL_LEVEL,
+        self::ADVANCED_SKILL_LEVEL,
+    ];
+
```
I believe I know what you mean here but would like to quick chat about it

TammyTee commented:
Hey @damiani ! I corrected all of the edits excluding the two I haven't marked as resolved. Let me know when you have a quick sec to chat about the things you noted so I can knock these last two things out.

Thanks for reviewing this!! 😄

TammyTee commented:
@damiani The module listing changes have been completed

**What changed:** 
* created a `ModuleListing` component for ease of switching between designs when a user is a guest or logged in
* added a toggle for logged in users to filter between the modules they're seeing
* added a badge to cards to indicate if a module is completed

**Screenshot**
![image](https://user-images.githubusercontent.com/8689444/84525917-6e1c7d00-acaa-11ea-9ab5-7c69d4a8d5df.png)

damiani commented on resources/js/components/Tabs/Tabs.vue:
```diff
@@ -75,12 +84,12 @@ export default {
             }
 
-            if($(window).width() >= 992) {
+            if(window.innerWidth >= 992) {
```
```suggestion
            if (window.innerWidth >= 992) {
```
